### PR TITLE
[Concept Entry] Rust Threads

### LINKED ISSUE #3879: [Concept Entry] Rust Threads
### Reason/inspiration (optional)

We would like a new concept entry for Rust about the threads. The entry should go in a new file in docs/content/rust/concepts/threads/threads.md

The entry should
- Have an introductory paragraph about the concept.
- Code block(s) displaying different aspects of the concept.
- A concluding header, Functions, future terms will be listed below this header

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by andersooi**
### Description

- Added new concept entry for Rust Threads
- Added new tag: Threads
- Closes #3879 

### Type of Change

- Adding a new entry
- Updating the documentation (Tags)

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/rust/concepts/threads/threads.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+---
+Title: ‘Rust Threads’
+Description: ‘Threads allow parts of a program to run concurrently.’
```
Comment (dakshdeepHERE) [inline content/rust/concepts/threads/threads.md:3]:
```suggestion
Description: ‘Threads allow parts of a program to run concurrently’
```

File: content/rust/concepts/threads/threads.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            println!("Spawned thread {}", i);
+            thread::sleep(Duration::from_millis(3));
+        }
+    });
+
+    for i in 1..4 {
+        println!("Main thread {}", i);
+        thread::sleep(Duration::from_millis(1));
+    }
+}
```
Comment (dakshdeepHERE) [inline content/rust/concepts/threads/threads.md:35]:
```suggestion
fn main() {
  thread::spawn(|| {
    for i in 1..10 {
        println!("Spawned thread {}", i);
        thread::sleep(Duration::from_millis(3));
      }
  });

  for i in 1..4 {
    println!("Main thread {}", i);
    thread::sleep(Duration::from_millis(1));
  }
}
```

File: content/rust/concepts/threads/threads.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+```
+
+The output is as follows:
```
Comment (dakshdeepHERE) [inline content/rust/concepts/threads/threads.md:38]:
```suggestion
The above example will give the following output:
```

File: content/rust/concepts/threads/threads.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+Spawned thread 9
+```
+
+Notice that the spawned threads reach 9 this time, even though the main thread has completed its execution.
+
+## Using `move` Closures with Threads
+
+Passing in the `move` keyword into the `thread::spawn()` function enables the safe transfer of ownership of a variable from one thread to another.
+
+Consider the following example without using the `move` keyword:
```
Comment (dakshdeepHERE) [inline content/rust/concepts/threads/threads.md:105]:
```suggestion
Passing in the `move` keyword into the `thread::spawn()` function enables the safe transfer of ownership of a variable from one thread to another. Consider the following example without using the `move` keyword:
```

File: content/rust/concepts/threads/threads.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+use std::thread;
+
+fn main() {
+    let greeting = String::from("hello");
+    let handle = thread::spawn(|| {
+        println!("{}", greeting);
+    });
+
+    handle.join().unwrap();
+}
```
Comment (dakshdeepHERE) [inline content/rust/concepts/threads/threads.md:117]:
```suggestion
use std::thread;

fn main() {
  let greeting = String::from("hello");
  let handle = thread::spawn(|| {
    println!("{}", greeting);
  });

  handle.join().unwrap();
}
```

File: content/rust/concepts/threads/threads.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+5 |     let handle = thread::spawn(|| {
+  |                                ^^ may outlive borrowed value `greeting`
+6 |         println!("{}", greeting);
+  |                        -------- `greeting` is borrowed here
+  |
+```
+
+Rust will attempt to borrow `greeting` into the spawned thread, but it cannot guarantee how long the spawned thread will run for. Rust’s borrowing rules do not allow a thread to outlive the data it is borrowing potentially.
+
+To solve this issue, pass in the `move` keyword as shown in the example below:
```
Comment (dakshdeepHERE) [inline content/rust/concepts/threads/threads.md:135]:
```suggestion
Rust will attempt to borrow `greeting` into the spawned thread, but it cannot guarantee how long the spawned thread will run for. Rust’s borrowing rules do not allow a thread to outlive the data it is borrowing potentially. To solve this issue, pass in the `move` keyword as shown in the example below:
```

File: content/rust/concepts/threads/threads.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+use std::thread;
+
+fn main() {
+    let greeting = String::from("hello");
+    let handle = thread::spawn(move || {
+        println!("{}", greeting);
+    });
+
+    handle.join().unwrap();
+}
```
Comment (dakshdeepHERE) [inline content/rust/concepts/threads/threads.md:147]:
```suggestion
use std::thread;

fn main() {
  let greeting = String::from("hello");
  let handle = thread::spawn(move || {
    println!("{}", greeting);
  });

  handle.join().unwrap();
}
```

File: content/rust/concepts/threads/threads.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+```
+
+The output is as follows:
```
Comment (dakshdeepHERE) [inline content/rust/concepts/threads/threads.md:150]:
```suggestion
The above example will give the following output:
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I have few comments otherwise this looks good 👍🏻

**Review by dakshdeepHERE — APPROVED** [review]
Looks good for second review 🚀

File: content/rust/concepts/threads/threads.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
+---
+Title: ‘Rust Threads’
+Description: ‘Threads allow parts of a program to run concurrently’
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Threads'
+CatalogContent:
+  - ‘learn-rust’
+  - 'paths/computer-science'
```
Comment (avdhoottt) [inline content/rust/concepts/threads/threads.md:10]: Add these slugs instead of this - 
```
- 'rust-for-programmers'
- 'paths/computer-science'
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@andersooi I left one change, please check them out.

**Pull request merged**